doc/SAI-Proposal-15-UOM.md

```diff
@@ -0,0 +1,229 @@
+Switch Abstraction Interface Change Proposal
+
+=====================
+Title    | Unified Object Model
+-------- | ---
+Authors  | Microsoft
+Status   | In review
+Type     | Standards track
+Created  | 04/12/2016
+SAI-Version | 0.9.5
+
+----------
+
+## Overview
+
+
+SAI is to provide unified API to configure switch ASIC resources. Every type of ASIC resources is represented by a type of SAI object with four operations: create, remove, set and get. With all these SAI objects being defined, the ASIC state can now be represented using a set of SAI objects, e.g., port, routes, vlan, nexthop.  
+
+This set of SAI objects (later referred as SAI view) can be used for many purposes. For example, we can use these SAI objects to capture current ASIC states and later used for restore the ASIC states after ASIC re-initialization. We can also compare two sets of SAI objects and find their differences and then make transition from one ASIC state for another ASIC state. 
+
+Most of SAI objects are now using Unified Object ID and CRUD API to manage them. However, some SAI objects are still having specific APIs to manage. For example, VLAN has add/remove ports API to add/remove ports from VLAN. These APIs complicates the operations needed to restore ASIC states from a SAI view, or transition from one SAI view to another SAI view, because the program needs to use these specific APIs to restore or make a transition. This also applies to port break out operation where specific attributes is used for such operation.

```

Comment:
What about the objects such as FDB, route and Neighbour which still dont have object_id associated and have to be handled thorough the key. In the warmboot proposal we have different getter/bulk getter APIs for each of those.